web/src/components/product/ProductSelectionProgress.jsx

```diff
@@ -21,152 +21,38 @@
-      variant: variant(progress),
-      isCurrent: isCurrent(progress),
-      description: description(progress)
-    };
-
-    if (properties.isCurrent) properties.icon = <Spinner />;
-
-    return properties;
-  };
-
-  // Emulates progress for product selection step.
-  const productProgress = () => {
-    if (!storageProgress.start) return { start: true, current: 1 };
-
-    return { start: true, current: 0 };
-  };
-
-  /** @todo Add aria-label to steps, describing its status and variant. */
-  return (
-    <ProgressStepper isCenterAligned>
-      <ProgressStep
-        id="product-step"
-        titleId="product-step-title"
-        {...stepProperties(productProgress())}
-      >
-        {selectedProduct.name}
-      </ProgressStep>
-      <ProgressStep
-        id="storage-step"
-        titleId="storage-step-title"
-        {...stepProperties(storageProgress)}
-      >
-        {_("Analyze disks")}
-      </ProgressStep>
-      <ProgressStep
-        id="software-step"
-        titleId="software-step-title"
-        {...stepProperties(softwareProgress)}
-      >
-        {_("Configure software")}
-      </ProgressStep>
-    </ProgressStepper>
-  );
-};
+import { useInstallerClient } from "~/context/installer";
 
 /**
  * @component
  *
  * Shows progress steps when a product is selected.
- *
- * @note Some details are hardcoded (e.g., the steps, the order, etc). The progress API has to be
- *  improved.
  */
 function ProductSelectionProgress() {
-  const { cancellablePromise } = useCancellablePromise();
-  const { manager, storage, software } = useInstallerClient();
   const { selectedProduct } = useProduct();
-  const [storageProgress, setStorageProgress] = useState({});
-  const [softwareProgress, setSoftwareProgress] = useState({});
+  const { manager } = useInstallerClient();
   const [status, setStatus] = useState();
 
   useEffect(() => {
     manager.getStatus().then(setStatus);
     return manager.onStatusChange(setStatus);
   }, [manager, setStatus]);
 
-  useEffect(() => {
-    const updateProgress = (progress) => {
-      if (progress.current > 0) progress.start = true;
-      setStorageProgress(p => ({ ...p, ...progress }));
-    };
-
-    cancellablePromise(storage.getProgress()).then(updateProgress);
-
-    return storage.onProgressChange(updateProgress);
-  }, [cancellablePromise, setStorageProgress, storage]);
-
-  useEffect(() => {
-    const updateProgress = (progress) => {
-      if (progress.current > 0) progress.start = true;
-      setSoftwareProgress(p => ({ ...p, ...progress }));
-      // Let's assume storage was started too.
-      setStorageProgress(p => ({ ...p, start: progress.start }));
-    };
-
-    cancellablePromise(software.getProgress()).then(updateProgress);
-
-    return software.onProgressChange(updateProgress);
-  }, [cancellablePromise, setSoftwareProgress, software]);
+  if (!selectedProduct) {
+    return;
+  }
 
   if (status === IDLE) return <Navigate to="/" replace />;
```

Comment:
Probably we could move this navigation to `App`:

~~~

    if (status === IDLE && location.pathname === "/products/progress") {
      return <Navigate to="/" />;
    }
~~~

So App would contain all the "routing" logic until we will improve the router usage.

Reply:
I see your point but 1) it is not part of this PR *and* I am scared of breaking the navigation again. I would keep it as it is until we fix the router usage.